drivers/entropy/entropy_renesas_ra.c

```diff
@@ -35,15 +33,27 @@ static int entropy_ra_rsip_trng_get_entropy(const struct device *dev, uint8_t *b
 	return 0;
 }
 
-static const struct entropy_driver_api entropy_ra_rsip_trng_api = {
-	.get_entropy = entropy_ra_rsip_trng_get_entropy,
+static const struct entropy_driver_api entropy_renesas_ra_api = {
+	.get_entropy = entropy_renesas_ra_get_entropy,
 };
 
-static int entropy_ra_rsip_trng_init(const struct device *dev)
+static int entropy_renesas_ra_init(const struct device *dev)
 {
 	HW_SCE_McuSpecificInit();
 	return 0;
 }
 
-DEVICE_DT_INST_DEFINE(0, entropy_ra_rsip_trng_init, NULL, NULL, NULL, PRE_KERNEL_1,
-		      CONFIG_ENTROPY_INIT_PRIORITY, &entropy_ra_rsip_trng_api);
+#define RENESAS_RA_ENTROPY_DEVICE_DEFINE(nodeid)                                                   \
+	DEVICE_DT_DEFINE(nodeid, entropy_renesas_ra_init, NULL, NULL, NULL, PRE_KERNEL_1,          \
+			 CONFIG_ENTROPY_INIT_PRIORITY, &entropy_renesas_ra_api)
+
+#define RENESAS_RA_RSIP_E51A_RNG_DEVICE_INIT_IF_OKAY(n)                                            \
+	COND_CODE_1(DT_NODE_HAS_STATUS_OKAY(DT_INST(n, renesas_ra_rsip_e51a_trng)),		   \
+		    (RENESAS_RA_ENTROPY_DEVICE_DEFINE(DT_INST(n, renesas_ra_rsip_e51a_trng))), ())
+
+#define RENESAS_RA_SCE7_RNG_DEVICE_INIT_IF_OKAY(n)                                                 \
+	COND_CODE_1(DT_NODE_HAS_STATUS_OKAY(DT_INST(n, renesas_ra_sce7_rng)),			   \
+		    (RENESAS_RA_ENTROPY_DEVICE_DEFINE(DT_INST(n, renesas_ra_sce7_rng))), ())
```

Comment:
Do you actually need those macros for every device?

Reply:
Done. I removed this macro because the SCE entropy driver is always single instance

Reply:
I suggest using preprocessor ifs instead of `COND_CODE` macros for readability (and it's how it's usually done from my experience).

Reply:
I fixed it